patches/tModLoader/Terraria/WorldGen.cs.patch

```diff
@@ -2041,14 +2041,30 @@
  							if (WallID.Sets.Conversion.Grass[wall] && wall != 69) {
  								tile.wall = 69;
  								SquareWallFrame(k, l);
-@@ -39585,6 +_,10 @@
+ 								NetMessage.SendTileSquare(-1, k, l);
+ 							}
+-							else if (TileID.Sets.Conversion.JungleGrass[type] && type != 661) {
+-								tile.type = 661;
+-								SquareTileFrame(k, l);
+-								NetMessage.SendTileSquare(-1, k, l);
+-							}
+ 							else if (WallID.Sets.Conversion.Stone[wall] && wall != 3) {
+ 								tile.wall = 3;
+ 								SquareWallFrame(k, l);
+@@ -39585,6 +_,16 @@
  								NetMessage.SendTileSquare(-1, k, l);
  							}
  
 +							SkipWallCorruptionConversion:
 +							if (!convertTile)
 +								goto SkipTileCorruptionConversion;
 +
++							if (TileID.Sets.Conversion.JungleGrass[type] && type != 661) {
```

Comment:
Here too, add `// Moved from above` to the end of this line so we can track why this edit is here.